togl/linuxwin/glmgrbasics.cpp

```diff
@@ -1,4684 +1,4686 @@
+	{ 0x00040000, "Mesa3DFX" },
+
+	VE( TERMVALUE )
+};
+
+
+//===============================================================================
+// decode helper funcs
+
+char	s_glmStrScratch[65536];
+int		s_glmStrCursor = 0;
+
+const char *	GLMDecode( GLMThing_t thingtype, unsigned long value )
+{
+	const GLMValueEntry_t *table = NULL;
+	//char			isflags = 0;
+
+	switch( thingtype )
+	{
+		case	eD3D_DEVTYPE:	table = g_d3d_devtypes;
+		break;
+		
+		case	eD3D_FORMAT:	table = g_d3d_formats;
+		break;
+			
+		case	eD3D_RTYPE:		table = g_d3d_rtypes;
+		break;
+			
+		case	eD3D_USAGE:		table = g_d3d_usages;
+		break;
+		
+		case 	eD3D_RSTATE:	table = g_d3d_rstates;
+		break;
+		
+		case	eD3D_SIO:		table = g_d3d_opcodes;			
+		break;
+		
+		case	eD3D_VTXDECLUSAGE:	table = g_d3d_vtxdeclusages;
+		break;
+		
+		case	eCGL_RENDID:	table = g_cgl_rendids;
+		break;
+			
+		case	eGL_ERROR:		table = g_gl_errors;
+		break;
+		
+		case	eGL_ENUM:		table = g_gl_enums;
+		break;
+
+		case	eGL_RENDERER:	table = g_gl_renderers;
+		break;
+
+		default:
+			GLMStop();
+			return "UNKNOWNTYPE";
+		break;
+	}
+	
+	if (table)
+	{
+		while( table->value != TERMVALUE )
+		{
+			if (table->value == value)
+			{
+				return table->name;
+			}
+			table++;
+		}
+	}
+	return "UNKNOWN";
+}
+
+const char	*GLMDecodeMask( GLMThing_t kind, unsigned long value )
+{
+	// if cursor to scratch buffer is within 1K of EOB, rewind
+	// nobody is going to decode 63K of flag string values in a single call..
+	
+	// this means that strings returned by this function have a short lifetime.. print them and do not save the pointer..
+	
+	if ( (sizeof(s_glmStrScratch) - s_glmStrCursor) < 1000 )
+	{
+		s_glmStrCursor = 0;
+	}
+	
+	char	*start = &s_glmStrScratch[ s_glmStrCursor ];
+	char	*dest = start;
+	char	first = 1;
+	
+	DWORD mask = static_cast<DWORD>(1L<<31);
+	while(mask)
+	{
+		if (mask & value)
+		{
+			sprintf(dest,"%s%s", (first) ? "" : "|", GLMDecode( kind, value&mask ) );
+			first = 0;
+			
+			dest += strlen(dest);	// leaves dest pointing at the end null
+		}
+		mask >>= 1;
+	}
+	s_glmStrCursor = (dest - s_glmStrScratch) + 1;	// +1 so the next decoded flag set doesn't land on the ending null
+	return start;
+	
+}
+
+#undef VE
+#undef TERMVALUE
+
+//===============================================================================
+
+bool	GLMDetectOGLP( void )
+{
+	bool result = false;
+#if defined( OSX ) && defined( CGLPROFILER_ENABLE )
+	GLint forceFlush;
+	CGLError error = CGLGetParameter(CGLGetCurrentContext(), kCGLCPEnableForceFlush, &forceFlush);
+	result = error == 0;
+	if (result)
+	{
+		// enable a breakpoint on color4sv
+		int oglp_bkpt[3] = { kCGLFEglColor4sv, kCGLProfBreakBefore, 1 };
+		
+		CGLSetOption( kCGLGOEnableBreakpoint, (GLint)oglp_bkpt );			
+	}
+	
+#endif
+	return result;
+}
+
+
+// from http://blog.timac.org/?p=190
+
+#ifndef _WIN32
+	#include <stdbool.h>  
+#endif
+#include <sys/types.h>  
+#ifndef _WIN32
+	#include <unistd.h>
+#endif

```

Comment:
Why?

Reply:
Not changed from our commits, either changed in master or a git bug.